tcp/server.go

```diff
@@ -49,35 +57,36 @@ func getTCProot(root string) string {
 	return root
 }
 
-//log Error if one occures
-func checkErr(err error) {
-	if err != nil {
-		qbin.Log.Errorf("TCP Server Error: %s", err.Error())
-	}
-}
-
 //handle incomming TCP Connection
 func handleClient(connTCP net.Conn, root string) {
+	qbin.Log.Debug("Handeling incomming connection...")
 	//set timeout to 5 seconds, limit requestlength to 1MB, close connection on programm close
-	connTCP.SetReadDeadline(time.Now().Add(5 * time.Second))
+	connTCP.SetReadDeadline(time.Now().Add(1500 * time.Millisecond))
 	result := make([]byte, 1024)
 	defer connTCP.Close()
 
 	result, err := ioutil.ReadAll(connTCP)
-	checkErr(err)
+	if err != nil {
+		qbin.Log.Errorf("TCP-read error: %s", err)
+		connTCP.Write([]byte("Ups, something went wrong."))
+		return
+	}
 	handleMsgProcessing(connTCP, string(result), root)
 
 }
 
 func handleMsgProcessing(connTCP net.Conn, msg string, root string) {
 	//length = 0 -> user already closed connection
 	if len(msg) == 0 {
+		qbin.Log.Debug("The TCP message was empty")
```

Comment:
Please send a error back to the client here instead of leaving the connection hanging.